test/common/wait.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"bufio"
+	"io"
+	"strings"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// WaitForObject waits for object to match the desired status.
+func WaitForObject(tl TestLogger, namespace, name string, objectGetter func(namespace, name string) (pkgruntime.Object, error), desired pkgruntime.Object, interval, timeout time.Duration, equivalent func(actual, desired pkgruntime.Object) bool) {
```

Comment:
Is that possible to use less arguments for this function? It is too many arguments to read and reuse it.
For example, the function is used in test context, `interval` and `timeout` are consistent in my simple check. It can be removed from the argument list.

Reply:
Since you're creating this file, how about putting it in `test/e2e/framework` instead?  `test/common` is a legacy of `test/integration` and `test/e2e` needing to share code, but `test/integration` no longer exists and pretty much everything in this path should end up in `test/e2e/framework`.